tests/lib/tools/report-mongodb

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import sys
+
+from pymongo import MongoClient
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--host', help="Hostname or ip for the mongo database",
+                        **environ_or_required('MONGO_HOST'))
+    parser.add_argument('--port', help="Port for the mongo database",
+                        **environ_or_required('MONGO_PORT'))
+    parser.add_argument('--db-name', help="Name for the database used to insert")
+    parser.add_argument('--db-collection', help="Name for the collection used to insert")
+    parser.add_argument("value", help="value to insert in the mongodb")
+
+    return parser
+
+
+def environ_or_required(key):
+    if os.environ.get(key):
+        return {'default': os.environ.get(key)}
+    else:
+        return {'required': True}
+
+
+def connect(host, port, db_name):
+    client = MongoClient(host=host, port=int(port))
+    if not db_name in client.list_database_names():
+        print("report-mongodb: invalid db name, it does not exist in mongodb")
+        sys.exit(1)
+    return client[db_name]
+
+
+def _get_json(value):
+    try:
+        return json.loads(value)
+    except ValueError as e:
+        return None
+
+
+def insert(db, col, value):
+    if not col in db.list_collection_names():
+        print("report-mongodb: invalid db collection, it does not exist in db")
+        sys.exit(1)
+
+    json_value = _get_json(value)
+    if not json_value:
+        print("report-mongodb: invalid value, it does not contain a valid json format")
+        sys.exit(1)
+
+    db[col].insert_one(json_value)
+    print("report-mongodb: json value inserted in db")
```

Comment:
Could you paste an example of the json output we would be inserting into mongodb?

Also, for someone unfamiliar with mongodb, can you explain how is this json blob going to be treated by mongodb and if later will we be able to sum up failures of same tests?

Reply:
Hi, I just added an example of the data which is being sent. This is just a tool to report, then we will have another tool to retrieve data with aggregations, but initially, the tool is just reporting errors to mongo which stores json values in a collection called spread_errors which is contained in the snapd database. 
The idea is to allow in the future, sending another info to the database using a different collection.